sites/cheerpx/src/content/docs/12-reference/00-CheerpX-Linux-create.md

```diff
@@ -0,0 +1,213 @@
+
+### `activityInterface`
+
+```ts
+activityInterface?: ActivityInterface;
+```
+
+This option allows monitoring of CPU and device activity within CheerpX, useful for handling state changes and resource management.
+
+### `bridgeURL`
+
+```ts
+bridgeURL?: string;
+```
+
+This option specifies the URL of a bridge server for communication between the CheerpX environment and external servers.
+
+Example:
+
+```js
+const cx = await CheerpX.Linux.create({
+	bridgeURL: "https://yourbridgeurl.com/api",
+});
+```
+
+### Device Configuration Options for CheerpX
+
+CheerpX supports various types of devices that can be configured as overlayDevice. Here’s how you can create them:
+
+- HttpBytesDevice (bytes): The default choice for loading filesystem images via HTTP. Suitable for most web-hosted files.
+- CloudDevice (block): Optimized for use with Cloudflare, enhancing performance and reliability through cloud storage solutions.
+- GitHubDevice (split): Ideal for projects integrated with GitHub Actions, allowing direct file loading from GitHub repositories.
+- IDBDevice: Provides persistent local storage using the browser's IndexedDB, perfect for applications requiring data retention across sessions.
+
+Example: Creating an Overlay Device
+
+```js
+const overlayDevice = await CheerpX.OverlayDevice.create(
+	await CheerpX.HttpBytesDevice.create("https://yourserver.com/image.ext2"),
+	await CheerpX.IDBDevice.create("block1"),
+);
+```
+
+### Using Different Device Types in Mounts
+
+CheerpX supports various device types that can be mounted and accessed like filesystems within the CheerpX environment.
+
+#### IDBDevice
+
+`CheerpX.IDBDevice` allows you to create a persistent storage device backed by IndexedDB.
+
+Example:
+
+```js
+const idbDevice = await CheerpX.IDBDevice.create("dbName");
+
+const cx = await CheerpX.Linux.create({
+	mounts: [{ type: "tree", path: "/files", dev: idbDevice }],
+});
+```
+
+This setup creates a virtual filesystem at `/files` that is backed by IndexedDB.
+
+#### WebDevice
+
+`CheerpX.WebDevice` allows you to mount a directory from your local server.
+
+Example:
+
+```js
+const webDevice = await CheerpX.WebDevice.create("/path/to/local/directory");
+
+const cx = await CheerpX.Linux.create({
+	mounts: [{ type: "tree", path: "/app", dev: webDevice }],
+});
+```
+
+This mounts the specified local directory at `/app` in the CheerpX environment.
+
+#### DataDevice
+
+`CheerpX.DataDevice` provides a simple in-memory filesystem.
+
+Example:
+
+```js
+const dataDevice = await CheerpX.DataDevice.create();
+
+const cx = await CheerpX.Linux.create({
+	mounts: [{ type: "tree", path: "/data", dev: dataDevice }],
+});
+```
+
+This creates an in-memory filesystem mounted at `/data`.
+
+### `dataDevice.writeFile`
+
+CheerpX provides a method to write data to new files within the mounted devices. This utility is limited to creating files at the root level of the device.
```

Comment:
"CheerpX.DataDevice provides a method ...", as this is the only device that provides such a utility.